javascript/ql/src/semmle/javascript/frameworks/Vue.qll

```diff
@@ -0,0 +1,361 @@
+     * those from extended objects and mixins.
+     */
+    abstract DataFlow::Node getOwnOption(string name);
+
+    /**
+     * Gets the node for option `name` for this instance, including those from
+     * extended objects and mixins.
+     */
+    DataFlow::Node getOption(string name) {
+      result = getOwnOption(name)
+      or
+      exists(DataFlow::SourceNode extendsVal | extendsVal.flowsTo(getOwnOption("extends")) |
+        result = extendsVal.(DataFlow::ObjectLiteralNode).getAPropertyWrite(name).getRhs()
+        or
+        exists(ExtendedVue extend |
+          MkExtendedVue(extendsVal) = extend and
+          result = extend.getOption(name)
+        )
+      )
+      or
+      exists(DataFlow::ArrayCreationNode mixins, DataFlow::ObjectLiteralNode mixin |
+        mixins.flowsTo(getOwnOption("mixins")) and
+        mixin.flowsTo(mixins.getAnElement()) and
+        result = mixin.getAPropertyWrite(name).getRhs()
+      )
+    }
+
+    /**
+     * Gets the node for the `data` option object of this instance.
+     */
+    DataFlow::Node getData() {
+      exists(DataFlow::SourceNode data | data.flowsTo(getOption("data")) |
+        result = data
+        or
+        // a constructor variant is available for all instance definitions
+        result.(DataFlow::SourceNode).flowsTo(data.(DataFlow::FunctionNode).getAReturn())
+      )
+    }
+
+    /**
+     * Gets the node for the `template` option of this instance.
+     */
+    DataFlow::Node getTemplate() { result = getOption("template") }
+
+    /**
+     * Gets the node for the `render` option of this instance.
+     */
+    DataFlow::Node getRender() { result = getOption("render") }
+
+    /**
+     * Gets the node for the `methods` option of this instance.
+     */
+    DataFlow::Node getMethods() { result = getOption("methods") }
+
+    /**
+     * Gets the node for the `computed` option of this instance.
+     */
+    DataFlow::Node getComputed() { result = getOption("computed") }
+
+    /**
+     * Gets a node for a member of the `methods` option of this instance.
+     */
+    private DataFlow::Node getAMethod() {
+      exists(DataFlow::SourceNode methods |
+        methods.flowsTo(getMethods()) and
+        result = methods.getAPropertyWrite().getRhs()
+      )
+    }
+
+    /**
+     * Gets a node for a member of the `computed` option of this instance that matches `kind` ("get" or "set").
+     */
+    private DataFlow::Node getAnAccessor(string kind) {
+      exists(DataFlow::SourceNode computedObj, DataFlow::SourceNode computedVal |
+        computedObj.flowsTo(getComputed()) and
+        computedVal.flowsTo(computedObj.getAPropertyWrite().getRhs())
+      |
+        result = computedVal and kind = "get"
+        or
+        result = computedVal.getAPropertyWrite(kind).getRhs()
+      )
+    }
+
+    /**
+     * Gets the node for the `name` getter of the `computed` option of this instance.
+     */
+    private DataFlow::Node getGetter(string name) {
```

Comment:
Isn't this the same as `getAnAccessor("get")`?

Reply:
Ah, nevermind, I see it now. There is no way to bind `name` when using `getAnAccessor`.